### PR TITLE
Update and rename AUTHORS to INITIAL_AUTHORS.md

### PR DESCRIPTION
**Pull request by CohenCyril**
##### Motivation for this change

@gares 
##### Things done/to do

- ~added corresponding entries in `CHANGELOG_UNRELEASED.md` (do not edit former entries)~
- ~added corresponding documentation in the headers~

##### Automatic note to reviewers

Read [this Checklist](https://github.com/math-comp/math-comp/wiki/Checklist-for-following,-reviewing-and-playing-with-a-PR#checklist-for-reviewing-a-pr) and make sure there is a milestone.

### CODE REVIEW

File: INITIAL_AUTHORS.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+Here are the initial authors of this libary, other contributors can be found in
+the [CHANGELOG.md](https://github.com/math-comp/math-comp/blob/master/CONTRIBUTING.md)
```
Comment (gares) [inline INITIAL_AUTHORS.md:2]: wrong link?

Comment (CohenCyril) [reply INITIAL_AUTHORS.md:2]:
```suggestion
the [CHANGELOG.md](https://github.com/math-comp/math-comp/blob/master/CHANGELOG.md)
```

**Comment by gares** [comment]
I wonder of you can add a link to https://github.com/math-comp/mathcomp-history-before-github

**Comment by CohenCyril** [comment]
> I wonder of you can add a link to https://github.com/math-comp/mathcomp-history-before-github

Why not, but how do you want to introduce it?

File: INITIAL_AUTHORS.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+Here are the initial authors of this libary, other contributors can be found in
```
Comment (gares) [inline INITIAL_AUTHORS.md:1]:
```suggestion
Here are the initial authors of this library which contributed to its development before the project moved to Github (the version control [history before github](https://github.com/math-comp/mathcomp-history-before-github) can still be accessed). Recent contributors can be found in
```

**Pull request merged**